scripts/lib/ts-generator.js

```diff
@@ -244,6 +245,24 @@ TSGenerator.prototype.generateSafeShapeName = function generateSafeShapeName(nam
     return name;
 };
 
+TSGenerator.prototype.importExtraDependency = function importExtraDependency() {
+    var code = '';
+    for (var importStatement in this.extraDependency) {
+        if (this.extraDependency[importStatement]) {
+            code += importStatement;
+        }
+    }
+    return code;
+}
+
+TSGenerator.prototype.extraDependencyFromShape = function extraDependencyFromShape(modelShape) {
+    var extraDependencyList = [];
+    if(JSON.stringify(modelShape).indexOf("\"streaming\":true") >= 0) {
```

Comment:
You should walk the shape and check the property explicitly. Models can have documentation that might include that string.